hardware_interface/include/eps.h

```diff
@@ -57,53 +57,74 @@ typedef struct __attribute__((packed)) {
     uint32_t uptimeInS;
-    uint32_t bootCnt; // system startup count
-    uint32_t wdt_gs_time_left; //seconds
+    uint32_t bootCnt;          // system startup count
+    uint32_t wdt_gs_time_left; // seconds
     uint32_t wdt_gs_counter;
-    uint16_t mpptConverterVoltage[4]; //mV
-    uint16_t curSolarPanels[8]; //mA
-    uint16_t vBatt; //mV
-    uint16_t curSolar; //mA
-    uint16_t curBattIn; //mA
-    uint16_t curBattOut; //mA
-    uint16_t curOutput[18]; //mA
-    uint16_t AOcurOutput[2]; //mA
-    uint16_t OutputConverterVoltage[8]; //mV
+    uint16_t mpptConverterVoltage[4];   // mV
+    uint16_t curSolarPanels[8];         // mA
+    uint16_t vBatt;                     // mV
+    uint16_t curSolar;                  // mA
+    uint16_t curBattIn;                 // mA
+    uint16_t curBattOut;                // mA
+    uint16_t curOutput[18];             // mA
+    uint16_t AOcurOutput[2];            // mA
+    uint16_t OutputConverterVoltage[8]; // mV
     uint8_t outputConverterState;
-    uint32_t outputStatus; //18-bits
-    uint32_t outputFaultStatus; //18-bits
+    uint32_t outputStatus;      // 18-bits
+    uint32_t outputFaultStatus; // 18-bits
     uint16_t protectedOutputAccessCnt;
-    uint16_t outputOnDelta[18]; //seconds
-    uint16_t outputOffDelta[18]; //seconds
+    uint16_t outputOnDelta[18];  // seconds
+    uint16_t outputOffDelta[18]; // seconds
     uint8_t outputFaultCnt[18];
-    int8_t temp[14]; //1-4 �MPPT converter temp, 5-8 �output converter temp, 9 �on-board battery temp, 10�12 �external battery pack temp, 13-14 -output expander temp
-    uint8_t battMode; //0 �critical, 1 �safe, 2 �normal, 3 �full
-    uint8_t mpptMode; //0 �HW, 1 �manual, 2 �auto, 3 �auto with timeout
-    uint8_t batHeaterMode; //0 �manual, 1 �auto
-    uint8_t batHeaterState; //0 �off, 1 �on
+    int8_t temp[14];       // 1-4 MPPT converter temp, 5-8 output converter temp, 9 on-board battery temp, 10-12
+                           // external battery pack temp, 13-14 -output expander temp
+    uint8_t battMode;      // 0 critical, 1 safe, 2 normal, 3 full
+    uint8_t mpptMode;      // 0 HW, 1 manual, 2 auto, 3 auto with timeout
+    uint8_t batHeaterMode; // 0 manual, 1 auto
+    uint8_t batHeaterState;   // 0 off, 1 on
     uint16_t PingWdt_toggles; // Total number of power channel toggles caused by failed ping watchdog
     uint8_t PingWdt_turnOffs; // Total number of power channel offs caused by failed ping watchdog
 };
 
-enum eps_mode {
-    critical = 0,
-    safe = 1,
-    normal = 2,
-    full = 3
+struct __attribute__((packed)) eps_startup_telemetry {
+    uint8_t cmd;
+    int8_t status;
+    double timestamp; // with ms precision
+    uint32_t last_reset_reason_reg; //Last reset reason register value, see below
+    uint32_t bootCnt;          // total system boot count
+    uint8_t FallbackConfigUsed;
+    uint8_t rtcInit;
+    uint8_t rtcClkSourceLSE;
+//  uint8_t flashAppInit;

```

Comment:
Is this not part of the telemetry? If not, remove this.

Reply:
we have multiple revisions of EPS ICD, this line is in one revision but not in others, let's keep it commented out for now since I don't see harm in keeping it as a reminder